templates/commands/render/action_include.go

```diff
@@ -36,36 +36,32 @@ func actionInclude(ctx context.Context, inc *spec.Include, sp *stepParams) error
 }
 
 func includePath(ctx context.Context, inc *spec.IncludePath, sp *stepParams) error {
+	skipPaths, err := processPaths(inc.Skip, sp.scope)
+	if err != nil {
+		return err
+	}
 	skip := make(map[string]struct{}, len(inc.Skip))
-	for _, s := range inc.Skip {
-		skipRelPath, err := parseAndExecuteGoTmpl(s.Pos, s.Val, sp.scope)
-		if err != nil {
-			return err
-		}
-		skip[skipRelPath] = struct{}{}
+	for _, s := range skipPaths {
+		skip[s.Val] = struct{}{}
 	}
 
-	for i, p := range inc.Paths {
-		// Paths may contain template expressions, so render them first.
-		walkRelPath, err := parseAndExecuteGoTmpl(p.Pos, p.Val, sp.scope)
-		if err != nil {
-			return err
-		}
+	incPaths, err := processPaths(inc.Paths, sp.scope)
+	if err != nil {
+		return err
+	}
 
+	asPaths, err := processPaths(inc.As, sp.scope)
+	if err != nil {
+		return err
+	}
+
+	for i, p := range incPaths {
 		// During validation in spec.go, we've already enforced that either:
 		//  - len(inc.As) == 0
 		//  - len(inc.As) == len(inc.Paths)
-		as := walkRelPath
-		if len(inc.As) > 0 {
-			as, err = parseAndExecuteGoTmpl(inc.As[i].Pos, inc.As[i].Val, sp.scope)
-			if err != nil {
-				return err
-			}
-		}
-
-		relDst, err := safeRelPath(p.Pos, as)
-		if err != nil {
-			return err
+		as := p.Val
+		if len(asPaths) > 0 {
+			as = asPaths[i].Val
 		}
```

Comment:
Is this correct? Can we guarantee that `len(asPaths) >= len(incPaths)`? Or should this be `len(asPaths) > i`?

Reply:
Updated comment assertion. See validation function in spec.go for IncludePath.